zkevm-circuits/src/evm_circuit/execution/error_oog_create2.rs

```diff
@@ -0,0 +1,362 @@
+        let size = block.rws[step.rw_indices[2]].stack_value();
+        self.address
+            .assign(region, offset, Some(address.to_le_bytes()))?;
+        self.size.assign(region, offset, Some(size.to_le_bytes()))?;
+
+        self.salt.assign(
+            region,
+            offset,
+            Some(block.rws[step.rw_indices[3]].stack_value().to_le_bytes()),
+        )?;
+
+        let address_high = address_high::value::<F>(address.to_le_bytes());
+        assert_eq!(
+            address_high,
+            F::zero(),
+            "address overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.address_in_range_high
+            .assign(region, offset, address_high)?;
+        let size_high = address_high::value::<F>(size.to_le_bytes());
+        assert_eq!(
+            size_high,
+            F::zero(),
+            "size overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.size_in_range_high.assign(region, offset, size_high)?;
+
+        let address_value = address_low::value(address.to_le_bytes());
+        let size_value = address_low::value(size.to_le_bytes());
+        let expanded_address = address_value
+            .checked_add(size_value)
+            .expect("address overflow u64");
+        assert!(
+            expanded_address < (1u64 << (N_BYTES_MEMORY_ADDRESS * 8)),
+            "expanded address overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.expanded_address_in_range.assign(
+            region,
+            offset,
+            F::from(expanded_address),
+            F::from(1u64 << (N_BYTES_MEMORY_ADDRESS * 8)),
+        )?;
+
+        let minimum_word_size = self.minimum_word_size.assign(region, offset, size_value)?;
+        let memory_expansion_gas = self
+            .memory_expansion
+            .assign(region, offset, step.memory_word_size(), [expanded_address])?
+            .1;
+
+        let constant_gas_cost = opcode.constant_gas_cost().0;
+
+        self.insufficient_gas.assign(
+            region,
+            offset,
+            F::from(step.gas_left),
+            F::from(6 * minimum_word_size + memory_expansion_gas + constant_gas_cost),
+        )?;
+
+        self.rw_counter_end_of_reversion.assign(
+            region,
+            offset,
+            Value::known(F::from(call.rw_counter_end_of_reversion as u64)),
+        )?;
+
+        self.restore_context
+            .assign(region, offset, block, call, step, 6)?;
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::test_util::CircuitTestBuilder;
+    use eth_types::{bytecode, word, Bytecode, ToWord, Word};
+    use mock::{
+        eth,
+        test_ctx::{helpers::account_0_code_account_1_no_code, LoggerConfig},
+        TestContext, MOCK_ACCOUNTS,
+    };
+
+    struct TestCase {
+        bytecode: Bytecode,
+        gas: Word,
+    }
+
+    #[test]
+    fn test() {
```

Comment:
have a look at other tests.. May not be a good name

Reply:
tests all cases, what could be a better name?

Reply:
test_oog_create2